analyzers/src/SonarAnalyzer.VisualBasic/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -53,5 +56,30 @@ public static bool IsPartOfBinaryNegationOrCondition(this SyntaxNode node)
 
         public static string FindStringConstant(this SyntaxNode node, SemanticModel semanticModel) =>
             FindConstantValue(node, semanticModel) as string;
+
+        public static ControlFlowGraph CreateCfg(this SyntaxNode block, SemanticModel semanticModel)
+        {
+            var operation = semanticModel.GetOperation(block);
+            var rootSyntax = operation.RootOperation().Syntax;
+            var cfg = ControlFlowGraph.Create(rootSyntax, semanticModel);
+            if (block is LambdaExpressionSyntax)
+            {
+                // We need to go up and track all possible enclosing lambdas and other FlowAnonymousFunctionOperations
+                var cfgFlowOperations = cfg.FlowAnonymousFunctionOperations();  // Avoid recomputing for ancestors that do not produce FlowAnonymousFunction
+                foreach (var node in block.AncestorsAndSelf().TakeWhile(x => x != rootSyntax).Reverse())
+                {
+                    if (cfgFlowOperations.SingleOrDefault(x => x.WrappedOperation.Syntax == node) is var flowOperation && flowOperation.WrappedOperation != null)
+                    {
+                        cfg = cfg.GetAnonymousFunctionControlFlowGraph(flowOperation);
+                        cfgFlowOperations = cfg.FlowAnonymousFunctionOperations();
+                    }
+                    else if (node == block)
+                    {
+                        return null;    // Lambda syntax is not always recognized as FlowOperation for invalid syntaxes
```

Comment:
VB.NET CFG creation can actually not-generate nested CFG for invalid syntax. See UT `CreateCfg_UndefinedSymbol_ReturnsNull_VB`. This was a reason why I made the C# more defensive as well.